Framework/Beamline/inc/MantidBeamline/ComponentInfo.h

```diff
@@ -48,6 +48,8 @@ class MANTID_BEAMLINE_DLL ComponentInfo {
                 const std::vector<std::pair<size_t, size_t>> &ranges);
   std::vector<size_t> detectorIndices(const size_t componentIndex) const;
   size_t size() const;
+  bool operator==(const ComponentInfo& other) const;
+  bool operator!=(const ComponentInfo& other) const;
```

Comment:
@SimonHeybrock yes, I'll move these to `DetectorInfo::isEquivalent` as part of the positions/rotations. I did have this in place to support some of the unit tests. I'm going to leave it in for now.